routes/admin_tutorial.go

```diff
@@ -0,0 +1,213 @@
+	PublicKeyBase58Check string
+	IsRemoval bool
+	IsWellKnown bool
+	JWT       string
+}
+
+type AdminGetTutorialCreatorsRequest struct {
+	ResponseLimit int
+	JWT       string
+}
+
+
+type AdminGetTutorialCreatorResponse struct {
+	UndiscoveredPublicKeysBase58Check []string
+	WellKnownPublicKeysBase58Check []string
+}
+
+func (fes *APIServer) AdminUpdateTutorialCreator(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := AdminUpdateTutorialCreatorRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("AdminUpdateTutorialCreator: Problem parsing request body: %v", err))
+		return
+	}
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("AdminUpdateTutorialCreator: error getting utxoview: %v", err))
+		return
+	}
+
+	var userMetadata *UserMetadata
+	userMetadata, err = fes.getUserMetadataFromGlobalState(requestData.PublicKeyBase58Check)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("AdminUpdateTutorialCreator: error getting usermetadata for public key: %v", err))
+		return
+	}
+
+	if !requestData.IsRemoval && ((userMetadata.IsFeaturedTutorialWellKnownCreator && requestData.IsWellKnown) || (userMetadata.IsFeaturedTutorialUndiscoveredCreator && !requestData.IsWellKnown)) {
+		var currentList string
+		var newList string
+		if requestData.IsWellKnown {
+			newList = "well known creators"
+			currentList = "undiscovered creators"
+		} else {
+			currentList = "well known creators"
+			newList = "undiscovered creators"
+		}
+		_AddBadRequestError(ww, fmt.Sprintf("AdminUpdateTutorialCreator: cannot add creator to %v, user already exists in %v", newList, currentList))
+		return
+	}
+
+	pkid := utxoView.GetPKIDForPublicKey(userMetadata.PublicKey)
+	if pkid == nil {
+		_AddBadRequestError(ww, fmt.Sprintf("AdminUpdateTutorialCreator: No PKID found for public key: %v", requestData.PublicKeyBase58Check))
+		return
+	}
+	var prefix []byte
+	wellKnownPrefix := GlobalStateKeyWellKnownTutorialCreators(pkid.PKID)
+	undiscoveredPrefix := GlobalStateKeyUndiscoveredTutorialCreators(pkid.PKID)
+	if requestData.IsWellKnown {
+		prefix = wellKnownPrefix
+	} else {
+		prefix = undiscoveredPrefix
+	}
+
+	// If adding a new creator, make sure they aren't already in the database
+	if (!requestData.IsRemoval) {
+		creatorExistsInWellKnown, err := fes.GlobalStateGet(wellKnownPrefix)
+		if err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"AdminUpdateTutorialCreator: Error processing GlobalStateGet: %v", err))
+			return
+		}
+
+	creatorExistsInUndiscovered, err := fes.GlobalStateGet(wellKnownPrefix)
+		if err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"AdminUpdateTutorialCreator: Error processing GlobalStateGet: %v", err))
+			return
+		}
+
+		if creatorExistsInWellKnown != nil || creatorExistsInUndiscovered != nil {
```

Comment:
We should check if creatorExistsInWellKnown before getting creatorExistsInUndiscovered to save a trip to the db